src/main/python/rlbot/agents/hivemind/python_hivemind.py

```diff
@@ -0,0 +1,157 @@
+from typing import Dict
+
+import queue
+import time
+
+from rlbot.botmanager.agent_metadata import AgentMetadata
+from rlbot.botmanager.bot_helper_process import BotHelperProcess
+from rlbot.messages.flat import MatchSettings
+from rlbot.utils.game_state_util import GameState
+from rlbot.utils.logging_utils import get_logger
+from rlbot.utils.rendering.rendering_manager import RenderingManager
+from rlbot.utils.structures.ball_prediction_struct import BallPrediction
+from rlbot.utils.structures.bot_input_struct import PlayerInput
+from rlbot.utils.structures.game_data_struct import GameTickPacket, FieldInfoPacket
+from rlbot.utils.structures.game_interface import GameInterface
+
+
+class PythonHivemind(BotHelperProcess):
+
+    # Terminology:
+    # hivemind - main process controlling the drones.
+    # drone - a bot under the hivemind's control.
+
+    def __init__(self, agent_metadata_queue, quit_event, options):
+        super().__init__(agent_metadata_queue, quit_event, options)
+
+        # Sets up the logger. The string is the name of your hivemind.
+        # This is configured inside your config file under hivemind_name.
+        self.logger = get_logger(options['name'])
+
+        # Give a warning if exec_path was given.
+        if 'exec_path' in options:
+            self.logger.warning(
+                """An exec_path was given, but this is a PythonHivemind, and so it is ignored.
+                Try subclassing SubprocessHivemind if you want to use an executable."""
+            )
```

Comment:
I love your good error messages!

Reply:
Your compliment made me realise this is actually outdated!
I no longer pass the `exec_path` as an option. Instead I set it as a class attribute when inheriting from `SubprocessHivemind`. Fixing this ASAP.